src/tile_id.cpp

```diff
@@ -58,3 +63,24 @@ size_t std::hash<TileId>::operator()(TileId const& tile_id) const
   boost::hash_combine(seed, tile_id.zoom);
   return seed;
 }
+
+std::string TilesToQuadkey(int x, int y, int z) 
+{
+  std::stringstream quadkey;
+  for (int i = z; i > 0; i--) {
+    char b = '0';
+    int mask = 1 << (i - 1);
+    if ((x & mask) != 0) 
+    {
+      b++;
+    }
+    if ((y & mask) != 0) 
+    {
+      b++;
+      b++;
+    }
+    quadkey << b;
+  }
+  // std::cout << "quadkey: " << std::endl;
+  return quadkey.str();
+}
```

Comment:
You removed the line ending from this file, would you mind putting it back? Now this line doesn't have an ending...

Reply:
OK. Modified in the new commit.